flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/ExpressionParser.scala

```diff
@@ -324,6 +332,8 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
     suffixToDate |
     // expression for log
     suffixLog |
+    // expression for ordering
+    suffixAsc | suffixDesc |
```

Comment:
@walterddr when you add the keyword `ASC` and allow it as a suffix, my ASCII scalar function test case : 

```
testAllApis(
      'f0.ascii(),
      "f0.ascii()",                  //here
      "ASCII(f0)",
      "84")
```

will report this error:

```
org.apache.flink.table.api.ExpressionParserException: Could not parse expression at column 7: string matching regex `(?i)\Qas\E' expected but `i' found F0.ascii()
```

After commenting `suffixAsc |` , my test case will pass, but your test will throw an exception.

I think I should rename `ASCII` to other name, such as `ASICII`?  cc @fhueske @twalthr @xccui

Reply:
Naming the function `asicii` does not sound right to me. I'd never look for this name if I'd was looking for ASCII conversion. There should be a proper fix for this. 
Please move this discussion to the ascii function PR.